casr/src/bin/casr-afl.rs

```diff
@@ -100,26 +100,35 @@ fn main() -> Result<()> {
                 .required(false)
                 .num_args(1..)
                 .last(true)
-                .help("Add \"-- ./gdb_fuzz_target <arguments>\" to generate additional crash reports with casr-gdb (e.g., test whether program crashes without sanitizers)"),
+                .help("Add \"-- ./gdb_fuzz_target <arguments>\" to generate additional crash reports with casr-gdb \
+                      (for compiled binaries, e.g., test whether program crashes without sanitizers), \"-- dotnet <arguments>\" \
+                      or \"-- mono <arguments>\" to triage C# crashes with additional options")
         )
         .get_matches();
 
     // Init log.
     util::initialize_logging(&matches);
 
-    let casr_san = util::get_path("casr-san")?;
-    let casr_gdb = util::get_path("casr-gdb")?;
-
-    let mut gdb_args = if let Some(argv) = matches.get_many::<String>("ARGS") {
+    let mut args = if let Some(argv) = matches.get_many::<String>("ARGS") {
         argv.cloned().collect()
     } else {
         Vec::new()
     };
 
-    if gdb_args.is_empty() && matches.get_flag("ignore-cmdline") {
+    if args.is_empty() && matches.get_flag("ignore-cmdline") {
         bail!("ARGS is empty, but \"ignore-cmdline\" option is provided.");
     }
 
+    // Get tool.
+    let mut tool = if matches.get_flag("ignore-cmdline")
+        && (args[0].ends_with("dotnet") || args[0].ends_with("mono"))
```

Comment:
This check repeats multiple times in code. Can you think how to revise the code for better readability?

Reply:
Maybe we can just have a flag for csharp and use it throughout the code...

Reply:
done, this check is provided only one time now inside the cycle